ush/python/pygfs/task/atmens_analysis.py

```diff
@@ -0,0 +1,332 @@
+
+        # get list of diag files to put in tarball
+        diags = glob.glob(os.path.join(self.task_config.DATA, 'diags', 'diag*nc4'))
+
+        logger.info(f"Compressing {len(diags)} diag files to {atmensstat}.gz")
+
+        # gzip the files first
+        logger.debug(f"Gzipping {len(diags)} diag files")
+        for diagfile in diags:
+            with open(diagfile, 'rb') as f_in, gzip.open(f"{diagfile}.gz", 'wb') as f_out:
+                f_out.writelines(f_in)
+
+        # open tar file for writing
+        logger.debug(f"Creating tar file {atmensstat} with {len(diags)} gzipped diag files")
+        with tarfile.open(atmensstat, "w") as archive:
+            for diagfile in diags:
+                diaggzip = f"{diagfile}.gz"
+                archive.add(diaggzip, arcname=os.path.basename(diaggzip))
+
+        # copy full YAML from executable to ROTDIR
+        logger.info(f"Copying {self.task_config.fv3jedi_yaml} to {self.task_config.COMOUT}")
+        src = os.path.join(self.task_config.DATA, f"{self.task_config.CDUMP}.t{self.task_config.cyc:02d}z.atmens.yaml")
+        dest = os.path.join(self.task_config.COMOUT, f"{self.task_config.CDUMP}.t{self.task_config.cyc:02d}z.atmens.yaml")
+        logger.debug(f"Copying {src} to {dest}")
+        yaml_copy = {
+            'mkdir': [self.task_config.COMOUT],
+            'copy': [[src, dest]]
+        }
+        FileHandler(yaml_copy).sync()
+
+        # Create UFS model readable atm increment file from UFS-DA atm increment
+        logger.info("Create UFS model readable atm increment file from UFS-DA atm increment")
+        self.jedi2fv3inc()
+
+    def clean(self):
+        super().clean()
+
+    @logit(logger)
+    def get_bkg_dict(self, task_config: Dict[str, Any]) -> Dict[str, List[str]]:
+        """Compile a dictionary of model background files to copy
+
+        This method constructs a dictionary of FV3 RESTART files (coupler, core, tracer)
+        that are needed for global atmens DA and returns said dictionary for use by the FileHandler class.
+
+        Parameters
+        ----------
+        task_config: Dict
+            a dictionary containing all of the configuration needed for the task
+
+        Returns
+        ----------
+        bkg_dict: Dict
+            a dictionary containing the list of model background files to copy for FileHandler
+        """
+        # NOTE for now this is FV3 RESTART files and just assumed to be fh006
+
+        bkgdir = [
+            os.path.join(task_config.DATA, 'bkg'),
+        ]
+        FileHandler({'mkdir': bkgdir}).sync()
+
+        # loop over ensemble members
+        bkglist = []
+        for imem in range(1, task_config.NMEM_ENKF + 1):
+            memchar = f"mem{imem:03d}"
+
+            # make run directory for member restart files
+            bkgdir = [
+                os.path.join(task_config.DATA, 'bkg', memchar)
+            ]
+            FileHandler({'mkdir': bkgdir}).sync()
```

Comment:
Instead of calling this for every member, consider creating a list and calling `FileHandler` once.

Reply:
Thank you @aerorahul for the good suggestion.  Change committed at [f902370](https://github.com/RussTreadon-NOAA/global-workflow/commit/f902370ad57b553bf4d4fa5861250ae23a68650c).